aws_lambda_powertools/utilities/event_handler/appsync.py

```diff
@@ -0,0 +1,145 @@
+from typing import Any, Dict
+
+from aws_lambda_powertools.utilities.data_classes import AppSyncResolverEvent
+from aws_lambda_powertools.utilities.typing import LambdaContext
+
+
+class AppSyncResolver:
+    """
+    AppSync resolver decorator
+
+    Example
+    -------
+
+    **Sample usage**
+
+        from aws_lambda_powertools.utilities.data_classes import AppSyncResolverEvent
+        from aws_lambda_powertools.utilities.event_handler import AppSyncResolver
+
+        app = AppSyncResolver()
+
+        @app.resolver(type_name="Query", field_name="listLocations", include_event=True)
+        def list_locations(event: AppSyncResolverEvent, page: int = 0, size: int = 10):
+            # Your logic to fetch locations
+            ...
+
+        @app.resolver(type_name="Merchant", field_name="extraInfo", include_event=True)
+        def get_extra_info(event: AppSyncResolverEvent):
+            # Can use "event.source" to filter within the parent context
+            ...
+
+        @app.resolver(field_name="commonField")
+        def common_field():
+            # Would match all fieldNames matching 'commonField'
+            ...
+    """
+
+    current_event: AppSyncResolverEvent
+    lambda_context: LambdaContext
```

Comment:
Maybe I missed, but if we're not a) exposing it to customers, or b) not reusing it across instantiations, then this could be a constructor private attribute.

Reply:
@heitorlessa  i was thinking this would be public and replace `include_event` and `include_context` (either or both)